subsys/task_wdt/task_wdt.c

```diff
@@ -199,7 +199,14 @@ int task_wdt_feed(int channel_id)
 	/* update task wdt kernel timer */
 	k_timer_user_data_set(&timer, (void *)next_channel_id);
 	k_timer_start(&timer, K_TIMEOUT_ABS_TICKS(next_timeout),
-		K_TIMEOUT_ABS_TICKS(next_timeout));
+		/* If the fallback hardware watchdog is used, also periodic
+		 * executions of the timer handler need to be scheduled
+		 * so that the hardware watchdog gets a chance to be fed
+		 * when its timeout turns out to be shorter than any active
+		 * channel timeout.
+		 */
+		IS_ENABLED(CONFIG_TASK_WDT_HW_FALLBACK) ?
+			K_MSEC(CONFIG_TASK_WDT_MIN_TIMEOUT) : K_NO_WAIT);
```

Comment:
Isn't it an issue that `next_channel_id` differs from `TASK_WDT_BACKGROUND_CHANNEL` in the repeated calls to the timer's expiry function?

Reply:
I think I agree with @pdgendt, it's probably not good to have a miss-match between channel ID and timeout. If `CONFIG_TASK_WDT_MIN_TIMEOUT` was configured bad (as it actually was in the sample) the `TASK_WDT_BACKGROUND_CHANNEL` should automatically become the `next_channel_id`.

Thinking twice, we should not need any periodic execution of the timer at all, so we could just always set the `period` to `K_NO_WAIT` or to `K_FOREVER` (which leads to the same result, but is more intuitive, in my opinion).

Reply:
> Thinking twice, we should not need any periodic execution of the timer at all, so we could just always set the `period` to `K_NO_WAIT` or to `K_FOREVER` (which leads to the same result, but is more intuitive, in my opinion).

This was also my first thought that periodic executions should not be used at all. But I saw that the hardware watchdog handling needed those, so I tried to align to the existing implementation. After looking at this again, I found that such approach is problematic - if you just comment out both calls to `task_wdt_feed()` in the sample, there will be no reset at all. Therefore, I decided to rework this a bit deeper.

I agree that `K_FOREVER` instead of  `K_NO_WAIT` is more intuitive when supplied as the 'period' parameter. I just saw that `K_NO_WAIT` was mainly used for that purpose in Zephyr. But I guess that's just because it simple replaced the value of 0 that had to be used earlier in such calls.